js/ui/marker.js

```diff
@@ -13,6 +13,7 @@ var Point = require('point-geometry');
  * @param {HTMLElement=} element DOM element to use as a marker (creates a div element by default)
  * @param {Object=} options
  * @param {PointLike=} options.offset The offset in pixels as a [`PointLike`](#PointLike) object to apply relative to the element's top left corner. Negatives indicate left and up.
+ * @param {boolean=} [options.perspective=false] Whether the marker should scale along with the map geometry.
```

Comment:
Perspective can mean many things. For example, in some cases it might be desirable to not only scale a marker but also tilt it to lie flat against the tilted map. (More ideas at mapbox/mapbox-gl-native#5245.)

In the meantime, merely scaling the marker would still be desirable functionality; how about a more descriptive term, like `scalesWithViewingDistance`?